Entity/Player/g_player.cpp

```diff
@@ -18,12 +20,148 @@ void Player::Update() {
 		m_pStateMachine->ChangeState(PlayerIdle::Instance());
 	}
 
+	UpdatePlayerModel(dt);
 	m_pStateMachine->Update();
 }
 
-Player::Player() : BaseGameEntity(0), m_pStateMachine(nullptr) {
+Player::Player(glm::vec3 initialPosition)
+	: BaseGameEntity(0), m_Forward({ 0.0, -1.0, 0.0 }), m_Side({ -1.0, 0.0, 0.0 }), m_RotationAngle(0.0),
+	  m_Velocity(0.0), m_AnimationState(ANIM_STATE_IDLE), m_CollisionInfo() {
 	m_pStateMachine = new StateMachine(this);
 	m_pStateMachine->SetCurrentState(PlayerIdle::Instance());
+	LoadModel("models/multiple_anims/multiple_anims.iqm", initialPosition);
+}
+
+void Player::LoadModel(const char* path, glm::vec3 initialPosition) {
+	// Load IQM Model
+	IQMModel iqmModel = LoadIQM(path);
+
+	// Convert the model to our internal format
+	m_Model = CreateModelFromIQM(&iqmModel);
+	m_Model.isRigidBody = false;
+	m_Model.position = initialPosition;
+	m_Model.scale = glm::vec3(22.0f);
+
+	for (int i = 0; i < m_Model.animations.size(); i++) {
+		EllipsoidCollider* ec = &m_Model.ellipsoidColliders[i];
+		ec->radiusA *= m_Model.scale.x;
+		ec->radiusB *= m_Model.scale.z;
+		ec->center = m_Model.position + glm::vec3(0.0f, 0.0f, ec->radiusB);
+		glm::vec3 scale = glm::vec3(1.0f / ec->radiusA, 1.0f / ec->radiusA, 1.0f / ec->radiusB);
+		ec->toESpace = glm::scale(glm::mat4(1.0f), scale);
+	}
+
+	SetAnimState(&m_Model, ANIM_STATE_WALK);
+}
+
+void Player::UpdateCamera(Camera* camera) {
+
+	// Fix camera position
+	camera->m_Pos.x = m_Model.position.x;
+	camera->m_Pos.y = m_Model.position.y;
+	camera->m_Pos.z = m_Model.position.z + 70.0f;
+	camera->m_Pos += (-m_Forward * 80.0f);
+	// m_RotationAngle should already have the information about if we want to move left or right
+	if (KeyPressed(SDLK_RIGHT) || KeyPressed(SDLK_LEFT)) {
+		camera->RotateAroundUp(m_RotationAngle);
+	}
+}
+void Player::Render(IRender* renderer, Camera* camera) {
+	HKD_Model* models[1] = { &m_Model };
+	renderer->SetActiveCamera(camera);
+	renderer->Render(camera, models, 1);
+
+	// Draw Debug Line for player veloctiy vector
+	EllipsoidCollider ec = m_Model.ellipsoidColliders[m_Model.currentAnimIdx];
+	Line velocityDebugLine = { Vertex(ec.center), Vertex(ec.center + 200.0f * m_Velocity) };
+	velocityDebugLine.a.color = glm::vec4(1.0f, 1.0f, 0.0f, 1.0f);
+	velocityDebugLine.b.color = velocityDebugLine.a.color;
+	renderer->ImDrawLines(velocityDebugLine.vertices, 2, false);
```

Comment:
there seems to be a bug where the debug line is not rendering... i am currently looking into it